### ISSUE TITLE
Story embed widget height

### ISSUE THREAD

**Issue by wendynorris** [denverfoundation/storybase#529]
Is it possible to responsively resize embeds rather than the display the blue scroll bar? 

![Floodlight test embed-WP3 4 2](https://f.cloud.github.com/assets/2329239/8648/5b5a2bb0-447d-11e2-8a19-80b5ec5853de.jpeg)

I imagine most users will either want to embed Floodlight stories into posts or into sidebars of varying widths too. 

The scroll bar is not attractive and I fear it will limit use. 

Tested in WP 3.4.2 on Firefox 17 and Safari 5.

**Comment by wendynorris** [comment]
After a couple more tests with different stories, it looks like the scroll bar issue is definitely related to the copy length of the summary and/or the length-wide ratio of the featured image. 

One potentially easier fix, which also addresses an unrelated Explorer page layout issue, is to force a character limit on summary copy. 

Is it possible to force proportional sizing of featured images contained in the embed, as we do on the Explore page?

**Comment by ghing** [comment]
@tekhneco It's not possible to do auto-heigh resising in a really good way. I did a lot of reading and it seems like that's a non-trivial problem, mostly because of security restrictions and cross domain issues.  We could probably get this working, but it might take a lot of time.

The scroll-bars aren't awesome, so the way to get rid of them is to change the height explicitly.  If you add a `data-height` attribute to the placeholder link, you can explicitly set the height of the IFRAME.  For example:

```
<a href="http://staging.floodlightproject.org/stories/mo-betta-market/" class="storybase-story-embed" data-height="800px">Mo' Betta Green Marketplace on Floodlight</a>
<!-- If you are embedding multiple stories on your page, include this script only once -->
<script src="http://staging.floodlightproject.org/static/js/widgets.min.js" type="text/javascript"></script>
```

Would make the embeded IFRAME 800px tall instead of the default 500px.

We could add setting this property to the embedding UI, but I think that we should look at adoption before we spend a lot of time on this.

Also, as I mentioned before, I think we could take steps to make the story summaries a more consistent (and less wordy) length.  I think there are both visual and communication issues that come from some of the longer summaries that bug me every time I see them.

**Comment by wendynorris** [comment]
Yep, completely agree on the copy length issue. 

As for auto-sizing, spending a lot time on modifying the code doesn't seem wise. I'm with you on that. But I wouldn't advise simply using adoption (or lack thereof) as the metric for determining priority. 

Web/Blog publishers are a picky bunch and embedding something unattractive or uncontrollable on their sites is almost always a deal breaker. 

It's probably a better UI/UX interview question for Bill to pose and dig into a bit. 

Perhaps adding a note about adding the data-height attribute under the embed code box would be a good compromise.

**Comment by jwirfs-brock** [comment]
I like adding the data-height attribute. Savvy users will be able to control it, and we can include a note for people who may not be familiar with it.

Another thing I noticed is that on some sites, when you click inside the box for the embed code, it automatically selects it all so you can just copy it (without first having to do a select all). Can we do that, or is it non-trivial?

I agree that enforcing a word-limit on the summaries is a good idea. We've discussed it several times, and there haven't been any objections, so I think we're probably ready to implement it. What happens to existing summaries that are longer than the new word limit?

**Comment by jwirfs-brock** [comment]
I've opened an issue, #530, for summary length-limit discussions.

**Comment by wendynorris** [comment]
Confirmed the embed script works in WP 3.5.

**Comment by jwirfs-brock** [comment]
@tekneco @ghing Are we ok to close this issue for now, now that the summary length limit has been opened as a separate ticket?

**Comment by ghing** [comment]
@jwirfs-brock, it depends if you want some kind of help message describing how to set the height of the share widget.  If that's the case, I'd leave this open and decide where the message should be and what the text should be.  Note that if you want to have a message below the embed code in the sharing widget, I think it should be very, very short.  We could also link to a longer description of this in the help page.

**Comment by jwirfs-brock** [comment]
@ghing  I also noticed that when you try to resize the container with the embed code,  you make it bigger than the sharing pop-up (see below). This looks kind of strange to me, but isn't something that should keep us from deploying the embed feature.

![embed_screenshot](https://f.cloud.github.com/assets/1020518/17788/62962808-4887-11e2-9b35-c1650dabbc3d.png)

**Comment by jwirfs-brock** [comment]
@ghing Also, did you add the data-height attribute? I'm not seeing it on dev or staging.

I like your suggestion about having a link below the embed code to a longer explanation on the help page. I'll work on the copy for that.

**Comment by wendynorris** [comment]
I think this needs to remain open until we deploy the help copy on the
data-height attribute section.

Wendy

**
_wendy norris http://about.me/wendynorris_
ceo + founder :: tekhne media, inc
a digital information and media collective
720 281 9378 :: mobile

*\* http://about.me/wendynorris
*
*

**Comment by ghing** [comment]
Decision: 
- show the default value of the `data-height` attribute in the embed code (for now)
- Add a link (that opens in a new window) to a view that lets the user specify the height of the widget (in a form) and preview the form with these attributes.  In the future this view could be used to further customize the embed widget.

**Comment by jwirfs-brock** [comment]
I'm not sure if this is working correctly. I tried to embed a story using the default settings. Here's the embed code:

```
<a href="http://floodlightproject.org/stories/planting-peace-in-whittier/" class="storybase-story-embed" data-height="500">Planting Peace in Whittier on Floodlight</a>
<!-- If you are embedding multiple stories on your page, include this script only once -->
<script src="http://floodlightproject.org/static/js/widgets.min.js" type="text/javascript"></script>
```

But when I published it on the Children's Corridor website, here's what it looked like:

![FL_embed](https://f.cloud.github.com/assets/1020518/72787/d3e27ea0-6022-11e2-9e94-5a7351a46512.png)

Changing the data-height didn't seem to change the size of the embedded widget. Am I missing something?

**Comment by ghing** [comment]
@jwirfs-brock Oops, the data-height attribute needs to be `500px` not `500`.  I'll fix this in production now, but you can manually update your embed code and it should work.

**Comment by jwirfs-brock** [comment]
@ghing Great! That worked.

**Comment by ghing** [comment]
Fixed in production as well.

**Comment by jwirfs-brock** [comment]
Re-visiting this, it looks like we still need...
- Help copy on how to change the data-height attribute
- To have the entire share box resize along with the embed-code box

**Comment by ghing** [comment]
I opened #761 to more clearly reflect the need to make it clearer for users how to change the height of the widget, addressing @jwirfs-brock concern.  Also note that the concern of "have the entire share box resize along with the embed-code box" is reflected in #545. I'm going to close this issue as we've addressed the fundamental ability of being able to adjust the height of the embed widget, even if it's not immediately clear to users.
